core/src/main/java/hudson/model/LoadStatistics.java

```diff
@@ -180,17 +249,71 @@ public Api getApi() {
     }
 
     /**
-     * Updates {@link #totalExecutors} and {@link #busyExecutors} by using
-     * the current snapshot value.
-     *
-     * {@link #queueLength} is updated separately via {@link LoadStatisticsUpdater} to
-     * improve the efficiency because we are counting this for all {@link LoadStatistics} at once.
+     * @deprecated use {@link #updateCounts(LoadStatisticsSnapshot)}
      */
+    @Deprecated
     protected void updateExecutorCounts() {
-        int t = computeTotalExecutors();
-        int i = computeIdleExecutors();
-        totalExecutors.update(t);
-        busyExecutors.update(t-i);
+        updateCounts(computeSnapshot());
+    }
+
+    /**
+     * Updates all the series from the current snapshot.
+     * @param current the current snapshot.
+     * @since FIXME
+     */
+    protected void updateCounts(LoadStatisticsSnapshot current) {
+        definedExecutors.update(current.getDefinedExecutors());
+        onlineExecutors.update(current.getOnlineExecutors());
+        connectingExecutors.update(current.getConnectingExecutors());
+        busyExecutors.update(current.getBusyExecutors());
+        idleExecutors.update(current.getIdleExecutors());
+        availableExecutors.update(current.getAvailableExecutors());
+        queueLength.update(current.getQueueLength());
+    }
+
+    /**
+     * Returns the {@link Node} instances that this statistic counts.
+     * @return the {@link Node}
+     * @since FIXME
+     */
+    protected abstract Iterable<Node> getNodes();
+
+    /**
+     * Returns {@code true} is the specified {@link SubTask} from the {@link Queue} should be counted.
+     * @param item the {@link SubTask}
+     * @return {@code true} IFF the specified {@link SubTask} from the {@link Queue} should be counted.
+     * @since FIXME
+     */
+    protected abstract boolean matches(SubTask item);
+
+    public LoadStatisticsSnapshot computeSnapshot() {
```

Comment:
This is a key method exposed to the rest of the world, so it should have javadoc.